src/sentry/static/sentry/app/components/dashboards/widgetQueryFields.tsx

```diff
@@ -117,53 +118,74 @@ function WidgetQueryFields({
       required
       stacked
     >
-      {fields.map((field, i) => (
-        <QueryFieldWrapper key={`${field}:${i}`}>
-          <QueryField
-            fieldValue={explodeField({field})}
-            fieldOptions={fieldOptions}
-            onChange={value => handleChangeField(value, i)}
-            filterPrimaryOptions={option => {
-              // Only validate functions for timeseries widgets.
-              if (!doNotValidateYAxis && option.value.kind === FieldValueKind.FUNCTION) {
+      {fields.map((field, i) => {
+        const fieldValue = explodeField({field});
+        return (
+          <QueryFieldWrapper key={`${field}:${i}`}>
+            <QueryField
+              fieldValue={fieldValue}
+              fieldOptions={fieldOptions}
+              onChange={value => handleChangeField(value, i)}
+              filterPrimaryOptions={option => {
+                // Only validate function parameters for timeseries widgets and
+                // world map widgets.
+                if (
+                  !doNotValidateYAxis &&
```

Comment:
The `!` doNot is a bit of a twist, but the other conditions would be more wordy if the variable was `validateYAxis` :shrug: